src/espidf.rs

```diff
@@ -105,47 +94,87 @@ impl Tools {
     }
 }
 
-/// Installer for the esp-idf source and tools.
+/// Information about a esp-idf source and tools installation.
+pub struct EspIdf {
+    /// The esp-idf repository.
+    pub esp_idf: git::Repository,
+    /// The binary paths of all tools concatenated with the system `PATH` env variable.
+    pub exported_path: OsString,
+}
+
+impl EspIdf {
+    pub fn detect_from_path() -> Result<Option<Self>> {
+        if cmd!["python", "idf.py", "--help"].is_ok() {
+            let version =
+                EspIdfVersion::try_from_env_var()?.ok_or_else(|| anyhow::anyhow!("TODO"))?;
+
+            let idf = Self {
+                esp_idf: version,
+                exported_path: env::var_os("PATH").unwrap_or_else(OsString::new),
+            };
+
+            Ok(Some(idf))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+/// EspIdfVersion can be either a managed or unmanaged one.
+///
+/// The managed one is represented by a GIT url and a git Ref. The URL is optional, and if not provided
+/// will default to the main ESP-IDF GitHub repository.
+///
+/// An unmanaged ESP-IDF version is represented by a user-provided local clone of ESP-IDF.
+///
+/// The main difference between managed and unmanaged ESP-IDF versions is reflected in their naming:
+/// - EspIdfVersion::Managed values are cloned locally by the Installer instance, inside its tooling installation directory.
+///   Consenquently, these ESP-IDF repository clones will disappar if the installation directory is deleted by the user.
+/// - EspIdfVersion::Unmanaged values are designating a user-provided, already cloned ESP-IDF repository which lives
+///   outisde the Installer's installation directory. It is only read by the Installer so as the required tooling
+///   for this ESP-IDF repository to be installed.
 #[derive(Debug, Clone)]
-pub struct Installer {
-    version: git::Ref,
-    git_url: Option<String>,
-    local_install_dir: Option<PathBuf>,
-    opts: InstallOpts,
-    tools: Vec<Tools>,
+pub enum EspIdfVersion {
+    Managed((Option<String>, git::Ref)),
+    Unmanaged(git::Repository),
```

Comment:
> 

While it can be turned into a `PathBuf`, I think you are mixing here Case 1 with Case 2 (see my general comment to your general comment as for what Case 1 and Case 2) means. A brief summary: `EspIdf::detect_from_path` (to be renamed to `EspIdf::try_from_env`) tries to detect an **activated** ESP-IDF tooling, which indeed implies an unmanaged, user-provided ESP-IDF. This is Case 1. However, we might also handle Case 2, which is the case where the user provides us an unmanaged, already-cloned ESP-IDF repo, however she has NOT activated the environment for that repo. In Case 2, there is no "detection logic" as there is no activated ESP-IDF environment, yet we still need to use an unmanaged ESP-IDF repo, but install the tooling for it ourselves.

Reply:
See my comment below.